### ISSUE TITLE
SPIFFS - fail to mount

### ISSUE THREAD

**Issue by nicolasdb** [nicolasdb/M5Stamp-C3U#7]
Since I'm not able to setup the SDcard module in #6 , I decided to try to use the SPIFFS partition. 

# with platform.io
I followed this https://randomnerdtutorials.com/esp32-vs-code-platformio-spiffs/
Build and upload partition files with success, but then impossible to mount the partition. 

#### platformio.ini
```
[env:esp32c3]
platform = 
	https://github.com/platformio/platform-espressif32.git#feature/arduino-upstream
platform_packages = 
	framework-arduinoespressif32 @ https://github.com/espressif/arduino-esp32.git#2.0.3
	platformio/tool-esptoolpy @ https://github.com/tasmota/esptool/releases/download/v3.2/esptool-v3.2.zip
framework = arduino
board = esp32-c3-devkitm-1
board_build.mcu = esp32c3
board_build.partitions = huge_app.csv
board_build.variant = esp32c3
board_build.f_cpu = 160000000L
board_build.f_flash = 80000000L
board_build.flash_mode = dio
board_build.arduino.ldscript = esp32c3_out.ld
build_unflags = 
	-DARDUINO_ESP32_DEV
build_flags = 
	-DARDUINO_ESP32C3_DEV
    -DARDUINO_USB_CDC_ON_BOOT=1         ; this and
    -DARDUINO_USB_MODE=1                ; this to enable "USB CDC" option
;lib_deps = 


monitor_speed = 115200
monitor_filters = time
```
based on https://github.com/leCandas/M5Stamp-C3

### the code is
```
#include "SPIFFS.h"
void setup() {
  Serial.begin(115200);  
  if(!SPIFFS.begin(true)){
    Serial.println("An Error has occurred while mounting SPIFFS");
    return;
  }
    File file = SPIFFS.open("/text.txt");
  if(!file){
    Serial.println("Failed to open file for reading");
    return;
  }
    Serial.println("File Content:");
  while(file.available()){
    Serial.write(file.read());
  }
  file.close();
}
void loop() {
}
```
#### Build
```
....
Found 41 compatible libraries
Scanning dependencies...
Dependency Graph
|-- SPIFFS @ 2.0.0
|   |-- FS @ 2.0.0
Building in release mode
Building SPIFFS image from 'data' directory to .pio/build/esp32c3/spiffs.bin
/text.txt
/Float.tx
============ [SUCCESS] Took 0.97 seconds ===================
```

#### Upload filesystem

```
Auto-detected Flash size: 4MB
Flash will be erased from 0x00310000 to 0x003fffff...
Compressed 983040 bytes to 1940...
Writing at 0x00310000... (100 %)
Wrote 983040 bytes (1940 compressed) at 0x00310000 in 8.7 seconds (effective 903.8 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...
================= [SUCCESS] Took 15.55 seconds ==============
```

#### Error in monitor:
```
15:31:09.884 > E (10395) SPIFFS: mount failed, -10025
15:31:09.884 > [ 10354][E][SPIFFS.cpp:114] format(): Formatting SPIFFS failed! Error: -1
15:31:09.889 > [ 10354][E][SPIFFS.cpp:89] begin(): Mounting SPIFFS failed! Error: -1
15:31:09.889 > An Error has occurred while mounting SPIFFS
```

**Comment by nicolasdb** [comment]
I tried to simplify the .ini like that : 
```
[env:esp32c3]
platform = 
	https://github.com/platformio/platform-espressif32.git#feature/arduino-upstream
platform_packages = 
	framework-arduinoespressif32 @ https://github.com/espressif/arduino-esp32.git#2.0.3
	platformio/tool-esptoolpy @ https://github.com/tasmota/esptool/releases/download/v3.2/esptool-v3.2.zip
framework = arduino
board = esp32-c3-devkitm-1
build_flags = 
     -DARDUINO_USB_CDC_ON_BOOT=1         ; this and
     -DARDUINO_USB_MODE=1                ; this to enable "USB CDC" option
monitor_speed = 115200
monitor_filters = time
```
I kept the "build_flags" needed to enable monitor on M5Stamp C3U

#### Monitor ERROR
```
15:54:44.617 > ESP-ROM:esp32c3-api1-20210207
15:54:44.618 > Build:Feb  7 2021
15:54:44.618 > rst:0x7 (TG0WDT_SYS_RST),boot:0xd (SPI_FAST_FLASH_BOOT)
15:54:44.618 > Saved PC:0x40049a42
15:54:44.619 > SPIWP:0xee
15:54:44.620 > mode:QIO, clock div:1
15:54:44.622 > load:0x3fcd6100,len:0x438
15:54:44.624 > ets_loader.c 78 
15:54:45.917 > ESP-ROM:esp32c3-api1-20210207
15:54:45.918 > Build:Feb  7 2021
15:54:45.918 > 
```

---
Then this:
```
[env:esp32c3]
platform = espressif32
board = esp32-c3-devkitm-1
upload_protocol = esptool
; lib_deps = 
monitor_speed = 115200
monitor_filters = time
```
from https://docs.platformio.org/en/latest/boards/espressif32/esp32-c3-devkitm-1.html

#### Build ERROR
```
riscv32-esp-elf-g++: error: unrecognized command line option '-mlongcalls'
*** [.pio/build/esp32c3/src/main.o] Error 1
========================================= [FAILED]
```

**Comment by nicolasdb** [comment]
Then this: 
```
[env:ESP32-C3]
platform = espressif32
board = esp32-c3-devkitm-1
framework = arduino
board_build.flash_mode = dio
monitor_speed = 115200
monitor_filters = time
```
#### Upload FAIL
```
esptool write_flash: error: argument <address> <filename>: [Errno 2] No such file or directory: 
'.platformio/packages/framework-arduinoespressif32/tools/sdk/esp32c3/bin/bootloader__80m.bin'
*** [upload] Error 2
```

> This is a bug in version 2.0.4 of the arduino-espressif32 framework. In order to work around it, use version 2.0.3 until this bux is fixed. You can do this by appending
`platform_packages = framework-arduinoespressif32 @ https://github.com/espressif/arduino-esp32.git#2.0.3`

https://techoverflow.net/2022/07/28/how-to-fix-platformio-no-such-file-or-directory-bootloader__80m-bin-bootloader__40m-bin/

**Comment by nicolasdb** [comment]
So back to: 

```
[env:ESP32-C3]
platform = espressif32
platform_packages = framework-arduinoespressif32 @ https://github.com/espressif/arduino-esp32.git#2.0.3
board = esp32-c3-devkitm-1
framework = arduino
board_build.flash_mode = dio
build_flags = 
     -DARDUINO_USB_CDC_ON_BOOT=1         ; this and
     -DARDUINO_USB_MODE=1                ; this to enable "USB CDC" option
monitor_speed = 115200
monitor_filters = time
```


#### Monitor
16:11:30.174 > E (15585) SPIFFS: mount failed, -10025
16:11:30.174 > [ 15543][E][SPIFFS.cpp:114] format(): Formatting SPIFFS failed! Error: -1
16:11:30.178 > [ 15543][E][SPIFFS.cpp:89] begin(): Mounting SPIFFS failed! Error: -1
16:11:30.179 > An Error has occurred while mounting SPIFFS

**Comment by nicolasdb** [comment]
![image](https://user-images.githubusercontent.com/12049360/190862056-ee4f3e06-ad31-44cb-961c-913d692b28c7.png)

![image](https://user-images.githubusercontent.com/12049360/185740229-c1256f64-246e-417f-8619-71b821c85c08.png)

**Comment by nicolasdb** [comment]
# with ARDUINO IDE
https://randomnerdtutorials.com/install-esp32-filesystem-uploader-arduino-ide

> 1) Go to the [releases page and click the ESP32FS-1.0.zip](https://github.com/me-no-dev/arduino-esp32fs-plugin/releases/) file to download.

**spoiler alert :** 
```
A fatal error occurred: This chip is ESP32-C3 not ESP32. Wrong --chip argument?
SPIFFS Upload failed!
```

Which lead me to https://github.com/lorol/arduino-esp32fs-plugin#installation

#### kind of working
No more errors, but the monitor only show **File Content:** from `Serial.println("File Content:");`
and nothing else. 

![image](https://user-images.githubusercontent.com/12049360/190862808-c4604cc4-2e71-48be-aa83-ffe4355c9f5e.png)

and my text.txt is not empty.

**Comment by nicolasdb** [comment]
Ok, the example SPIFFS_Test.ino is working. 
I guess the first time it is used, the partition is formated because of `#define FORMAT_SPIFFS_IF_FAILED true`

**Comment by nicolasdb** [comment]
Ok, totaly working within Arduino IDE :+1: 
![image](https://user-images.githubusercontent.com/12049360/190863933-d31895e7-c9b6-4001-9f4b-7b39c814684a.png)

**Comment by nicolasdb** [comment]
still looking for solution in platformio. 
https://github.com/platformio/platform-espressif32/issues/546

I tried `board_build.flash_mode = qio` but I just get other kind of errors. 

#### monitor error:
```
15:33:09.847 > ESP-ROM:esp32c3-api1-20210207
15:33:09.847 > Build:Feb  7 2021
15:33:09.848 > rst:0x7 (TG0WDT_SYS_RST),boot:0xd (SPI_FAST_FLASH_BOOT)
15:33:09.848 > Saved PC:0x40049a42
15:33:09.849 > SPIWP:0xee
15:33:09.850 > mode:QIO, clock div:1
15:33:09.852 > load:0x3fcd6100,len:0x438
15:33:09.853 > ets_loader.c 78 
15:33:11.147 > ESP-ROM:esp32c3-api1-20210207
15:33:11.147 > Build:Feb  7 2021
15:33:11.148 > ..... LOOP,LOOP,LOOP......
```

**Comment by nicolasdb** [comment]
## yus, it's working!
sometimes, you just need to wait for the problem to be solved with the new package 2.0.5 :smiley: 

`platform_packages = 
	framework-arduinoespressif32 @ https://github.com/espressif/arduino-esp32.git#2.0.5`
> https://github.com/leCandas/M5Stamp-C3/blob/main/platformio.ini
